### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds tests which cover releases test cases.
In particular, there are two new tests, one which checks the installation directory, and others which checks the packaging and compile an example.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/110/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/104/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/111/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/105/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/112/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/106/)

File: ci/linux/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+set(_c_flags "-fwrapv -fprofile-arcs -ftest-coverage")
+set(_cxx_flags "-fwrapv -fprofile-arcs -ftest-coverage")
+set(_exe_linker_flags "-fprofile-arcs -ftest-coverage")
+set(_shared_linker_flags "-fprofile-arcs -ftest-coverage")
+
+check_cxx_compiler_flag("-fprofile-abs-path" _have_fprofile_abs_path)
+if(_have_fprofile_abs_path)
+    set(_cxx_flags "${_cxx_flags} -fprofile-abs-path")
+endif()
+
```
Comment (BorjaOuterelo) [inline ci/linux/CMakeLists.txt]:
```suggestion
check_c_compiler_flag("-fprofile-abs-path" _have_c_fprofile_abs_path)
if(_have_c_fprofile_abs_path)
    set(_c_flags "${_c_flags} -fprofile-abs-path")
endif()

```

File: cmake/SuperBuild.cmake [hunk 2]
```diff
@@ -16,22 +16,26 @@ include(ExternalProject)
```
Comment (BorjaOuterelo) [inline cmake/SuperBuild.cmake]: Shall we consider other compilers?

 CMAKE_CACHE_ARGS
            -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
            -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}

File: src/cpp/datareader/DataReader.cpp [hunk 3]
```diff
@@ -237,30 +237,26 @@ void DataReader::read_task(
...
-
-            if (token_bucket.get_tokens(data.size()))
-            {
-                bool submessage_pushed = false;
-                do {
-                    submessage_pushed = read_cb(cb_args, data, std::chrono::milliseconds(PUSH_SUBMESSAGE_TIMEOUT));
-                } while (running_cond_ && !submessage_pushed);
-
-                if (submessage_pushed)
+                if (token_bucket.consume_tokens(data.size(), MAX_SLEEP_TIME))    /* code */
```
Comment (BorjaOuterelo) [inline src/cpp/datareader/DataReader.cpp]:
```suggestion
                if (token_bucket.consume_tokens(data.size(), MAX_SLEEP_TIME))
```

File: test/case/packaging/main.cpp [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#include <uxr/agent/Agent.hpp>
```
Comment (BorjaOuterelo) [inline test/case/packaging/main.cpp]: Is it enough just with an include?
Shall que call some real code here?

File: test/case/installation/InstallationTest.cmake [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+else()
+    if(NOT EXISTS ${INSTALL_PATH}/lib/${LIBRARY_NAME})
+        message(FATAL_ERROR "Library lib/${LIBRARY_NAME} not found.")
+    endif()
+endif()
+
+# Check executable.
+if(NOT EXISTS ${INSTALL_PATH}/bin/${EXECUTABLE_NAME})
+    message(FATAL_ERROR "Executable bin/${EXECUTABLE_NAME} not found.")
+endif()
```
Comment (BorjaOuterelo) [inline test/case/installation/InstallationTest.cmake]: Is there any way to test includes?

Comment (julianbermudez) [reply test/case/installation/InstallationTest.cmake]: In some sense, includes are tested checking the existence of the folder and through the example. Of course, we can be more verbose and check each include file. It is up to you.

File: test/case/packaging/main.cpp (hunk 4 above)
Comment (julianbermudez) [reply test/case/packaging/main.cpp]: Real code is tested at `AgentUnitTest` but we can add some API function calls. What do you think?

File: CMakeLists.txt [hunk 6]
```diff
@@ -58,12 +65,43 @@ set(UAGENT_CONFIG_TCP_MAX_BACKLOG_CONNECTIONS  100      CACHE STRING "Maximum TC
 set(UAGENT_CONFIG_SERVER_QUEUE_MAX_SIZE        32000    CACHE STRING "Maximum server's queues size.")
 set(UAGENT_CONFIG_CLIENT_DEAD_TIME             5000     CACHE STRING "Client dead time in milliseconds.")
 
+###############################################################################
+# Dependencies
+###############################################################################
+set(_deps "")
+
+set(_fastcdr_version 1.0.10)
+set(_fastcdr_tag v1.0.11)
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply CMakeLists.txt]: It is [an issue](https://github.com/eProsima/Fast-CDR/issues/62) in Fast-CDR.

File: CMakeLists.txt [hunk 7]
```diff
@@ -269,6 +302,11 @@ file(COPY ${PROJECT_SOURCE_DIR}/agent.refs
 ###############################################################################
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake/modules)
 
+if(UAGENT_BUILD_CI_TESTS)
+    include(CTest)
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply CMakeLists.txt]: Yes, it is true. I took out common factor.

File: ci/linux/CMakeLists.txt (hunk 1 above)
Comment (julianbermudez) [reply ci/linux/CMakeLists.txt]: Added!

File: cmake/SuperBuild.cmake (hunk 2 above)
Comment (julianbermudez) [reply cmake/SuperBuild.cmake]: Added!

File: src/cpp/datareader/DataReader.cpp (hunk 3 above)
Comment (julianbermudez) [reply src/cpp/datareader/DataReader.cpp]: Fixed!

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/117/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/111/)

File: test/case/installation/InstallationTest.cmake (hunk 5 above)
Comment (BorjaOuterelo) [reply test/case/installation/InstallationTest.cmake]: I think is enough with the folder.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/118/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/112/)

**Pull request merged**